Makefile

```diff
@@ -12,8 +12,31 @@
 #		environments may also work but are not tested.
 #			cd ardop\ARDOPC
 #			mingw32-make
+#
+#	`make test` which builds the executable and also runs some tests also
+#	requires installation of cmocka, which is not required for the default build.
+#		On Debian/Ubuntu this is easily installed with:
+#			sudo apt install libcmocka-dev
+#
+#		Package managers for other Linux distributions are also likely to
+#		provide easy installation of cmocka.
+#
+#		In the following description of how to install cmocka for Windows, a
+#		winlibs MinGW installation is assumed to be located at `C:\winlibs`
+#		If installed elsewhere, substitute the appropriate path.  Putting the
+#		cmocka files into the winlibs install directory avoids the need for further
+#		configuration.  This uses git (available from https://git-scm.com/downloads/win)
+#		to download the cmocka source code.
+#
+#		git clone https://git.cryptomilk.org/projects/cmocka.git
+#		cd cmocka
+#		mkdir build
+#		cd build
+#		cmake -G "MinGW Makefiles" -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX="C:\winlibs" ..
+#		cmake --build . --config Release
+#		cmake --install . --config Release
```

Comment:
I made some revisions to your cmake build instructions. Specifically, I replaced the direct invocations of `make` with `cmake` commands. This makes the instructions a little more `-G` generator agnostic. Can you take them for a spin and see if they work? (Or if you don't like them, I can put it back.)

Reply:
I tried your alternate instructions for building cmocka, and they seem to work as well as the earlier version with `mingw32-make`.  However, I think that using `mingw32-make`, which is just the name traditionally given to `make` for MinGW, is probably the better solution.  Except for this name change, it matches what appears in INSTALL.md of the cmocka repository.  Following those directions were how I came up with what I wrote earlier.  Deviating more than necessary from those instructions seems counterproductive.

Today I also installed and configured a new 64-bit OS to a card for a Raspberry Pi Zero 2 W.  I'll be using this for building/testing future ardopcf_arm_Linux_64 releases.  In the process, I was reminded that my earlier statement that `For Linux, the default build requires only gcc and libraries typically installed by default.` is incorrect, at least for some Raspios installs like this one.  It is necessary to also do `sudo apt install libasound2-dev`.  This should be added to the build instructions for Linux. 

So, With those two minor changes to the comments, I think your testing system is ready to merge into the develop branch.  

However, your example test case and/or the changes you made in ARDOPC.c are not OK.  Your tests pass on Windows 32-bit and 64-bit and on arm Linux 64-bit, but on arm Linux 32-bit test_compress_callsign() fails.  This also comes back to my earlier comment that "For the test cases, I would like to see comments in the test source files explaining why these test values are believed to be correct."  Without such comments, it seems clear that there is a problem in the source code since the 32-bit and 64-bit versions are producing different results, but it is not clear whether either of the results is correct.

Reply:
@cbs228: Assuming that you may not have access to an arm 32-bit Linux system to facilitate debugging this problem, I'll look back at it tomorrow.  I should also be able to run some exchanges between ardopcf and Ardop_Win to produce verified test cases.

Reply:
I've restored your `mingw32-make` comments and also updated the build instructions for Linux.

I have added a brief comment on source of truth values for compressed callsigns.

I do not have any ARM32 systems on hand. I could potentially emulate one, but that would take me some time. Can you provide the console log from the test failure? Maybe run it with the sanitizers to see if any of the arithmetic has gone wonky?

Reply:
The code in ARDOPC.c seems to be OK.  It is the test cases that are based on incorrect assumptions.  The output of CompressCallsign() is only 6 bytes, but test_compress_callsign() compares the result to 8 bytes expressed as a hex string.  Bytes 7 and 8 are never set, so their values are indeterminate.  So, what is surprising is that the values matched on 3 of the 4 systems tested.

Reply:
Also, does this ARM32 target use glibc or musl? alpine (with its musl libc) is very popular for those kinds of devices.

If the `HEAD` commit fails the tests, I would check and see if either `HEAD~1` or `HEAD~2` pass. If I introduced a regression, the most likely place is my `snprintf()` calls. I may have inadvertently used a glibc-ism or something that musl just doesn't support. I'll double-check these later.

Reply:
The ARM32 target is a Raspberry Pi with the default Raspios operating system.  I believe that these use libc.  As I wrote two comments back, I think that the only problem is that your test cases fir test_compress_callsign() should only be comparing 6 bytes (12 hex characters), but you are comparing 8 bytes (16 hex characters).

Reply:
Sure enough: I fail at counting. 900efa43b539714457c01d51629f795ada771c14 narrows the compressed callsign to 6 bytes and adds an assert against the check string length.

I also had to tweak the .gitignore paths slightly. They were erroneously ignoring test-case sources.

This is an uninitialized read bug. I thought ASAN and UBSAN protected against those kinds of things, but it does not. You need MSan for that, which isn't part of gcc [yet]. But good old valgrind detects the problem:

```log
valgrind test/ardop/test_ARDOPC 
==569809== Memcheck, a memory error detector
==569809== Copyright (C) 2002-2022, and GNU GPL'd, by Julian Seward et al.
==569809== Using Valgrind-3.21.0 and LibVEX; rerun with -h for copyright info
==569809== Command: test/ardop/test_ARDOPC
==569809== 
[==========] Running 3 test(s).
[ RUN      ] test_ascii_6bit
[       OK ] test_ascii_6bit
[ RUN      ] test_compress_callsign
==569809== Use of uninitialised value of size 8
==569809==    at 0x4ADB712: _itoa_word (_itoa.c:179)
==569809==    by 0x4AE59A3: __printf_buffer (vfprintf-process-arg.c:155)
==569809==    by 0x4B0B405: __vsnprintf_internal (vsnprintf.c:96)
==569809==    by 0x4AE27D5: snprintf (snprintf.c:31)
==569809==    by 0x10C4E8: assert_callsign_wireline_is (test_ARDOPC.c:44)
==569809==    by 0x10C661: test_compress_callsign (test_ARDOPC.c:83)
==569809==    by 0x4A78AAB: ??? (in /usr/lib/x86_64-linux-gnu/libcmocka.so.0.7.0)
==569809==    by 0x4A79192: _cmocka_run_group_tests (in /usr/lib/x86_64-linux-gnu/libcmocka.so.0.7.0)
==569809==    by 0x10C8CD: main (test_ARDOPC.c:127)
==569809== 
==569809== Conditional jump or move depends on uninitialised value(s)
==569809==    at 0x4ADB724: _itoa_word (_itoa.c:179)
==569809==    by 0x4AE59A3: __printf_buffer (vfprintf-process-arg.c:155)
==569809==    by 0x4B0B405: __vsnprintf_internal (vsnprintf.c:96)
==569809==    by 0x4AE27D5: snprintf (snprintf.c:31)
==569809==    by 0x10C4E8: assert_callsign_wireline_is (test_ARDOPC.c:44)
==569809==    by 0x10C661: test_compress_callsign (test_ARDOPC.c:83)
==569809==    by 0x4A78AAB: ??? (in /usr/lib/x86_64-linux-gnu/libcmocka.so.0.7.0)
==569809==    by 0x4A79192: _cmocka_run_group_tests (in /usr/lib/x86_64-linux-gnu/libcmocka.so.0.7.0)
==569809==    by 0x10C8CD: main (test_ARDOPC.c:127)
```

which is confirmed fixed now. Whenever we get around to establishing CI jobs, we should include a valgrind run in addition to an ASAN/UBSAN build.